### DISCUSSION TITLE
Updating multiple models in single component w/ validation

### DISCUSSION THREAD

**Discussion by jrousan** [livewire/livewire#5428] [1. Help]
I'm creating a controller-less type of application, as such many of my Livewire components are closer to full-page components that interact with multiple models (seemed simpler than nesting components--tell me if I'm wrong). 

So I have one page where one parent Model "Provider" lists many of its child Models (e.g., locations, contacts, users).  On this page I have multiple modals which can edit the information for each child Model.

I have it working as it should, but I feel as though the approach is "hacky."   I have a global rules() function because without that, no data will show in the first place for editing.

But because each modal deals with a different Model, I **only** want to validate the information in that modal. So I'm doubling up on the validation for e.g.,

```
    public function rules()
    {
        return [
            'provider.name' => 'required|string|unique:providers,name,' . $this->provider->id,
            'provider.code' => 'required|string|unique:providers,provider_code,' . $this->provider->id,
            'provider.phone_number' => 'required|string|unique:providers,phone_number,' . $this->provider->id,
            'provider.is_admin' => 'required|boolean',

            // Big list of validation 
       ];
    }

```

Then in the update method for this class, I need to pass in that same data, to avoid validating the rest of what is in rules()

```
    public function updateProvider()
    {
        $this->validate([
            'provider.name' => 'required|string|unique:providers,name,' . $this->provider->id,
            'provider.code' => 'required|string|unique:providers,provider_code,' . $this->provider->id,
            'provider.phone_number' => 'required|string|unique:providers,phone_number,' . $this->provider->id,
            'provider.is_admin' => 'required|boolean',
        ]);
        $this->provider->save();
        $this->showEditProviderModal = false;
    }
```

Is there a better way to do this?

Initially I just added the "sometimes" parameter and this worked because if the Model wasn't loaded yet, it would skip validation, and if it was loaded it just re-validated the already existing data (i.e., it wasn't empty so it passed validation).  This stopped working when I added a "change password" modal because the public variable `$new_password` would always run through the validation and cause it to not submit (even if I made it null to start with). 

Any thoughts?

**Comment by joshhanley** [comment]
@jrousan personally, I would use nested components, but only for the modals. So I would have my full page component that displays all the data/ different models, but then have a nested component for each of the modals you need to edit the different model types.

Sometimes nested components can be a pain if a user goes overboard with them, but I think they are the best solution in this instance.

That way your modal components are simple, easy to maintain and easy to test.

Once the modal finishes saving a model, it can then dispatch an event to tell the full page component to update it's displayed data.

Hope this helps!
